apollo-federation/src/sources/connect/json_selection/helpers.rs

```diff
@@ -265,4 +275,32 @@ mod tests {
         ", "
             "));
     }
+
+    #[test]
+    fn test_is_identifier() {
+        assert!(is_identifier("hello"));
+        assert!(is_identifier("hello_world"));
+        assert!(is_identifier("hello_world_123"));
+        assert!(!is_identifier("hello world"));
+        assert!(!is_identifier("hello-world"));
+        assert!(!is_identifier("123hello"));
+        assert!(!is_identifier("hello@world"));
+        assert!(!is_identifier("$hello"));
+        assert!(!is_identifier("hello$world"));
+        assert!(!is_identifier(" hello"));
+        assert!(!is_identifier("__hello_world  "));
```

Comment:
is `assert!(!is_identifier("_hello_world"))` valid? String starting with `_`

Reply:
Yes, but I'll add some tests.

Reply:
https://github.com/apollographql/router/pull/7419/commits/74b26d22201156f083b54de83de2496ff3d313b7